include/grpcpp/impl/codegen/call_op_set.h

```diff
@@ -455,18 +452,20 @@ class CallOpRecvMessage {
         got_message = false;
         recv_buf_.Clear();
       }
-    } else {
-      got_message = false;
-      if (!allow_not_getting_message_) {
-        *status = false;
+    } else if (hijacked_) {
+      if (!hijacked_recv_message_status_) {
+        FinishOpRecvMessageFailureHandler(status);
       }
```

Comment:
Maybe a comment explaining why a fall-through happens if it was hijacked and had a recv_message_status ?